exchanges/populate/src/populateExchange.ts

```diff
@@ -284,35 +416,58 @@ export const addFragmentsToQuery = (
         }
 
         const newSelections = possibleTypes.reduce((p, possibleType) => {
-          const typeFrags = activeTypeFragments[possibleType.name];
+          let typeFrags = activeTypeFragments[possibleType.name];
```

Comment:
so here we should look at the collectedFieldsByType rather than the fragments we've collected